examples/drupal_loadtest.rs

```diff
@@ -98,26 +98,26 @@ fn drupal_loadtest_front_page(client: &mut GooseClient) {
             client.set_failure();
         },
     }
-}
+} task!(drupal_loadtest_front_page, drupal_loadtest_front_page_task);
```

Comment:
I feel like rustfmt is going to have something to say about this.

If we are going for something like a python decorator then attribute macros are the answer. It would also be familiar to anyone who has used the async-trait macro. We could write one in this create if you want: it's just a bit of a pain.

Let me have breakfast and then I'll see what I can do about making a version that expands inline.

Reply:
Thanks!

There's a `goose_codegen/` that was intended for this purpose -- it probably makes sense to use that library, replacing the broken code that was there.

Reply:
Sorry for the delay: I forgot that I had a family video call scheduled ;-)

I was originally thinking about something like this: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=efa9597001e725286a62b5e4ed9746b7

you could potentially move more code into the macro, depending on your appetite for magic.

Reply:
Yes, that's fantastic! Much cleaner/simpler. I committed this change.

I plan to fix up the documentation, comments, comment tests, and use a proper prelude, then I'll get this merged -- I welcome further improvements, but at this point I think it's as usable as pre-async so no need to block it longer. (I'm mostly away from my computer today, so I likely won't get to it until early next week)

I'd be curious what other magic you feel is possible? If we don't have to add any unusual dependencies I'm definitely happy to add macro-magic that simplifies the writing ergonomics of load tests.

Reply:
doc-tests are finally working again locally -- I'll get this merged and then worry about the documentation and further optimizations/cleanups.